client/src/main/java/io/split/client/utils/Json.java

```diff
@@ -1,13 +1,32 @@
 package io.split.client.utils;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import java.lang.reflect.Type;
 
 /**
  * Created by adilaijaz on 6/10/16.
  */
 public class Json {
 
-    private static final Gson _json = new Gson();
+    private static final Gson _json = new GsonBuilder()
+            .serializeNulls()  // Send nulls
+            .registerTypeAdapter(Double.class, new JsonSerializer<Double>() {
+
+                // Send integers as such
+                @Override
+                public JsonElement serialize(Double src, Type typeOfSrc, JsonSerializationContext context) {
+                    if (src == src.longValue())
```

Comment:
@mredolatti this is not good. What are you trying to do here?

Reply:
Whenever Json parses an "Object" type, which happens to be a number, it serializes it as a Double, and adds a decimal point to it. so 123 becomes 123.0 (ref: https://github.com/google/gson/blob/master/gson/src/main/java/com/google/gson/internal/bind/ObjectTypeAdapter.java#L78), prior to sending it. This then becomes stringified in the BE, and it gets stored in elasticsearch as "123.0" . I'm not really sure how exp does the comparisons and whether this might be an issue or not, but our QoS fails for sure.

If we're 100% sure it won't be an issue for exp  we can always remove that and add extra logic in our QoS to test for equality under those conditions.